receiver/kafkametricsreceiver/receiver.go

```diff
@@ -33,8 +33,10 @@ const (
 	consumersScraperName   = "consumers"
 )
 
+type createKafkaScraper = func(context.Context, Config, *sarama.Config, *zap.Logger) (scraperhelper.ResourceMetricsScraper, error)
```

Comment:
I don't think you want the = as that creates a type alias which is slightly different.

```suggestion
type createKafkaScraper func(context.Context, Config, *sarama.Config, *zap.Logger) (scraperhelper.ResourceMetricsScraper, error)
```

Reply:
Thank you! I didn't realize I had that there. Which is why the lint was still failing even after creating a name type for the function. I really wish creating an alias like this was illegal since its almost never done as opposed to aliasing an existing type.